Carbon.Windows/Functions/New-CEventLog.ps1

```diff
@@ -0,0 +1,41 @@
+
+function New-CEventLog
+{
+    <#
+    .SYNOPSIS
+    Creates an event log.
+
+    .DESCRIPTION
+    The `New-CEventLog` is a drop-in replacement for the `New-EventLog` function that is available in Windows
+    PowerShell, but with support for PowerShell Core.
+
+    .EXAMPLE
+    New-CEventLog -LogName 'TestApp' -Source 'TestLog'
+
+    Demonstrates creating a `TestLog` event log and registers `TestApp` as a source for the log.
+    #>
+    [CmdletBinding(SupportsShouldProcess)]
+    param(
+        # The name of the log to be created.
+        [Parameter(Mandatory)]
+        [String] $LogName,
+
+        # The source of
+        [Parameter(Mandatory)]
+        [String[]] $Source
+    )
+
+    Set-StrictMode -Version 'Latest'
+    Use-CallerPreference -Cmdlet $PSCmdlet -Session $ExecutionContext.SessionState
+
+    foreach ($s in $Source)
+    {
+        if (-not [System.Diagnostics.EventLog]::SourceExists($s))

```

Comment:
Write an error if the source exists. This is PowerShell-y: write an error if you can't do what the verb of your function is expected to do.

Write an `Install-CEventLog` function that does not write an error if the event log if it doesn't exist.